modules/eks-managed-node-group/variables.tf

```diff
@@ -365,7 +383,9 @@ variable "taints" {
 variable "update_config" {
   description = "Configuration block of settings for max unavailable resources during node group updates"
   type        = map(string)
-  default     = {}
+  default = {
+    max_unavailable_percentage = 33
```

Comment:
Isn't it strange default? Shouldn't we ask users to provide it instead of having 33 here?

Reply:
this is what AWS recommends for a starting point. Maybe it doesn't need to be precisely `33`, `30` is ok as well. But it means that when a new AMI or new version or new launch template version is ready to be rolled out, it will update the nodes in a controlled fashion where at most 30% of your nodes could be unavailable. So if you have 10 nodes, at most 3 of those will be unavailable during the rollout (note its *at most - usually its less than that). And likewise for the self-managed node group, the [default is now 66%](https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2250/files#diff-c8cfd2bc6fecf4475b30f0c42777c0378c994df608bea50e9c38d5970ec83667R479) which equates back to this roughly 1/3 of your compute pool being unavailable during an upgrade cycle